async/aws-parameter-store-manager-async/src/main/java/co/com/bancolombia/secretsmanager/connector/AWSParameterStoreConnectorAsync.java

```diff
@@ -0,0 +1,94 @@
+package co.com.bancolombia.secretsmanager.connector;
+
+import co.com.bancolombia.secretsmanager.api.GenericManagerAsync;
+import co.com.bancolombia.secretsmanager.api.exceptions.SecretException;
+import co.com.bancolombia.secretsmanager.config.AWSParameterStoreConfig;
+import com.github.benmanes.caffeine.cache.AsyncCache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import reactor.core.publisher.Mono;
+import software.amazon.awssdk.auth.credentials.*;
+import software.amazon.awssdk.services.ssm.SsmAsyncClient;
+import software.amazon.awssdk.services.ssm.SsmAsyncClientBuilder;
+import software.amazon.awssdk.services.ssm.model.GetParameterRequest;
+
+import java.net.URI;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+
+public class AWSParameterStoreConnectorAsync implements GenericManagerAsync {
+
+    private final AWSParameterStoreConfig config;
+    private SsmAsyncClient client;
+    private AsyncCache<String, String> cache;
+    private Logger logger = Logger.getLogger("connector.AWSSecretManagerConnector");
+
+    public AWSParameterStoreConnectorAsync(AWSParameterStoreConfig config) {
+        this.config = config;
+        this.client = buildClient();
+        this.cache = initCache();
+    }
+
+    @Override
+    public Mono<String> getSecret(String secretName) {
+        return Mono.fromFuture(cache.get(secretName,
+                (s, executor) -> getSecretValue(secretName)
+                        .toFuture().toCompletableFuture()));
+    }
+
+    @Override
+    public <T> Mono<T> getSecret(String secretName, Class<T> cls) {
+        return Mono.error(new UnsupportedOperationException("Serialization doesn't apply for parameter store connector"));
+    }
```

Comment:
can we implement this method?

Reply:
No, because Parameter Store only handle String type